test/model/src/driver/kv_workload_runner_wt.cpp

```diff
@@ -517,6 +530,32 @@ kv_workload_runner_wt::do_operation(const operation::truncate &op)
       session->session(), table_uri(op.table_id), cursor1, cursor2, op.start, op.stop);
 }
 
+/*
+ * kv_workload_runner_wt::do_operation --
+ *     Execute the given workload operation in WiredTiger.
+ */
+int
+kv_workload_runner_wt::do_operation(const operation::wt_config &op)
+{
+    std::unique_lock lock(_connection_lock);
+
+    size_t l = op.value.size() + 1; /* In-memory size, including the NUL byte. */
+    const char *v = op.value.c_str();
+
+    if (op.type == "connection") {
+        if (l > sizeof(_state->connection_config))
+            throw model_exception("The connection config is too long");
+        memcpy(_state->connection_config, v, l);
+    } else if (op.type == "table") {
+        if (l > sizeof(_state->table_config))
+            throw model_exception("The connection config is too long");
+        memcpy(_state->table_config, v, l);
```

Comment:
Why the `memcpy`? Is it realistic to have `_state->table_config` be a `std::string` and simply replace the thing? Or does the pointer need to be immutable for some reason? (This isn't necessarily wrong, I just don't understand the intention.)

Reply:
The WT workload runner executes the workload in a child process, so that it can intentionally crash WT at different points, plus WT can crash at any point due to an assertion. The runner uses `_state`, which is backed by shared memory, to keep around any state between WT crashes and restarts.  And because `_state` is simply just a pointer to one block of shared memory, it must use only C types and be fully self-contained, so there is no option to use `std::string` or any such fancy types that may allocate memory on the heap.

The workload currently allows `wt_config` to run at any point during the workload, which requires its state to be a part of the shared memory state `_state`. Strictly speaking, I didn't have to do it that way as this feature is not currently used by the workload generator, but it was easier to do it that way by leveraging the existing state mechanism than to do something one-off.

Reply:
Ah, that makes sense. Thanks for the explanation!